datafusion/core/src/physical_optimizer/combine_partial_final_agg.rs

```diff
@@ -229,7 +230,7 @@ mod tests {
     fn partial_aggregate_exec(
         input: Arc<dyn ExecutionPlan>,
         group_by: PhysicalGroupBy,
-        aggr_expr: Vec<Arc<dyn AggregateExpr>>,
+        aggr_expr: Vec<Arc<AggregateFunctionExpr>>,
```

Comment:
If `AggregateFunctionExpr` is not cloned massively in the core workflow, we could remove `Arc`, similar to that we doesn't wrap `Arc` for `Expr`

Reply:
I could try this in a followup pr.